src/components/Verse/VerseAndTranslation.tsx

```diff
@@ -32,37 +59,56 @@ interface Props {
 
 const VerseAndTranslation: React.FC<Props> = (props) => {
   // If fixedFontScale is provided as a prop, use it; otherwise, get from hook (Redux)
-  const { fixedFontScale, ...restProps } = props;
+  const { fixedFontScale, chapter, ...restProps } = props;
   const {
     data,
     error,
     mutate,
     translationFontScale: reduxTranslationFontScale,
     quranTextFontScale: reduxQuranTextFontScale,
-  } = useVerseAndTranslation(restProps);
-  const { arabicVerseClassName, translationClassName, quranFont } = props;
+  } = useVerseAndTranslation({ ...restProps, chapter });
+
+  const {
+    data: chapterData,
+    error: chapterError,
+    isLoading: chapterIsLoading,
+    mutate: mutateChapter,
+  } = useChapter({
+    chapterIdOrSlug: chapter.toString(),
+  });
+
+  const handleRetry = useCallback(() => {
+    mutate();
+    mutateChapter();
+  }, [mutate, mutateChapter]);
 
-  if (error) return <Error error={error} onRetryClicked={mutate} />;
+  if (error || chapterError) {
+    return <Error error={error || chapterError} onRetryClicked={handleRetry} />;
+  }
 
-  if (!data) return <Spinner />;
+  if (!data || chapterIsLoading) return <Spinner />;
 
   return (
     <div className={styles.container}>
       {data?.verses?.map((verse) => (
         <div key={verse.verseKey} className={styles.verseContainer}>
-          <div className={classNames(styles.arabicVerseContainer, arabicVerseClassName)}>
+          <div className={classNames(styles.arabicVerseContainer, restProps.arabicVerseClassName)}>
             <PlainVerseText
-              quranFont={quranFont}
+              shouldShowTitle
+              quranFont={restProps.quranFont}
               words={getVerseWords(verse)}
               fontScale={fixedFontScale ?? reduxQuranTextFontScale}
             />
           </div>
-          <div className={classNames(styles.translationsListContainer, translationClassName)}>
+          <div
+            className={classNames(styles.translationsListContainer, restProps.translationClassName)}
+          >
             {verse.translations?.map((translation) => (
               <div key={translation.id} className={styles.translationContainer}>
                 <TranslationText
+                  chapterName={chapterData?.chapter?.nameComplex}
+                  reference={`${verse.chapterId}:${verse.verseNumber}`}
```

Comment:
This should be localized, we already have `toLocalizedVerseKey` method. This is how it looks in Arabic for example which is not localized!

<img width="1409" height="261" alt="Image" src="https://github.com/user-attachments/assets/6c271529-09f8-4d52-a64e-5db815901a67" />